app/error.tsx

```diff
@@ -0,0 +1,34 @@
+'use client' // Error components must be Client Components
+ 
+import { Button } from '@/components/ui/button'
+import { useEffect } from 'react'
+ 
+export default function Error({
+  error,
+  reset,
+}: {
+  error: Error & { digest?: string }
+  reset: () => void
+}) {
+  useEffect(() => {
```

Comment:
Unsure if we need this useEffect here (irrespective of vercel logging the error) or not --> u could just console.log(error) outside the effect and it would still print everytime the error changes as it would cause this component to rerender since error comes in as a prop.

Reply:
Stupid rule i use that is super handy --> If you are using a useEffect, don't.

There will come a point where you will need it (which is why it exists) but 99% of the time there is a better way to do it. Reading reacts own documentation on useEffect gave me a really goo dunderstanding on this.